jjb/include/pipelines/qpc-tests-pipeline.groovy

```diff
@@ -0,0 +1,529 @@
+// {release} {install_type}
+
+def qpc_version = "" // Better way for this?
+if ('{release}' == 'master') {{
+    qpc_version = "0.0.47"
+}} else {{
+    qpc_version = "{release}"
+}}
+
+
+def image_name = "quipucords:${{qpc_version}}"
+def tarfile = "quipucords.${{qpc_version}}.tar"
+def targzfile = "${{tarfile}}.gz"
+def install_tar = "quipucords.install.tar"
+def install_targzfile = "${{install_tar}}.gz"
+
+def getQPCVersion() {{
+    if ('{release}' == 'master') {{
+        return "0.0.47"
+    }} else {{
+        return "{release}"
+    }}
+}}
+
+/////////////////////////
+//// Setup Functions ////
+/////////////////////////
+def setupDocker() {{
+    sh """\
+    echo "OPTIONS=--log-driver=journald" > docker.conf
+    echo "DOCKER_CERT_PATH=/etc/docker" >> docker.conf
+    echo "INSECURE_REGISTRY=\\"--insecure-registry \${{DOCKER_REGISTRY}}\\"" >> docker.conf
+    sudo cp docker.conf /etc/sysconfig/docker
+    """.stripIndent()
+}}
+
+
+
+/////////////////////////////////////
+//// Configure & Start Functions ////
+/////////////////////////////////////
+def startQPCServer = {{
+    sh """\
+    echo 'startQPCServer'
+    pwd
+    ls -lah
+    sudo systemctl start docker # Ensure docker running
+    QPC_DB=\$(sudo docker ps | grep qpc-db || true)
+    QPC_CONTAINER=\$(sudo docker ps | grep ${{image_name}} || true)
+
+    # Stop containers if they are running before restarting or testing
+    if [ \"\$QPC_DB\" != '' ] || [ \"\$QPC_CONTAINER\" != '' ]
+    then
+        sudo docker rm \$(sudo docker stop \$(sudo docker ps -aq))
+    fi
```

Comment:
Is there any disadvantage to doing `sudo docker rm \$(sudo docker stop \$(sudo docker ps -aq))` all the time to reduce complexity here?

Reply:
Yep. If `docker rm` is run when there are no containers to stop, it breaks the pipeline. I could alternatively just run the `docker rm` command `if` there are containers. I decided to leave it like this though because 1) I already had this for something else I was doing in an earlier version, and 2) the if checks what I _actually_ care about (if `QPC_DB` or `$QPC_CONTAINER` containers exist).